code/modules/client/preference_setup/loadout/gear.dm

```diff
@@ -0,0 +1,139 @@
+	if(!islist(tags))
+		tags = list(tags)
+	fill_automatic_tags_on_item(src)
+
+/datum/gear_data
+	var/path
+	var/location
+	var/faction_requirement
+
+/datum/gear_data/New(var/path, var/location, var/faction)
+	src.path = path
+	src.location = location
+	src.faction_requirement = faction
+
+/datum/gear/proc/cant_spawn_item_reason(var/location, var/metadata, var/mob/living/carbon/human/human, var/datum/job/job, var/datum/preferences/prefs)
+	var/datum/gear_data/gd = new(path, location, faction)
+	for(var/datum/gear_tweak/gt in gear_tweaks)
+		if(metadata["[gt]"])
+			gt.tweak_gear_data(metadata["[gt]"], gd, human)
+		else
+			gt.tweak_gear_data(gt.get_default(), gd, human)
+
+	var/obj/spawning_item = gd.path
+	if(length(allowed_roles) && !(job.title in allowed_roles))
+		return "You cannot spawn with the [initial(spawning_item.name)] with your current job!"
+	if(!check_species_whitelist(human))
+		return "You cannot spawn with the [initial(spawning_item.name)] with your current species!"
+	if(gd.faction_requirement && (human.employer_faction != "Stellar Corporate Conglomerate" && gd.faction_requirement != human.employer_faction))
+		return "You cannot spawn with the [initial(spawning_item.name)] with your current faction!"
+	var/our_culture = text2path(prefs.culture)
+	if(culture_restriction && !(our_culture in culture_restriction))
+		return "You cannot spawn with the [initial(spawning_item.name)] with your current culture!"
+	var/our_origin = text2path(prefs.origin)
+	if(origin_restriction && !(our_origin in origin_restriction))
+		return "You cannot spawn with the [initial(spawning_item.name)] with your current origin!"
+	return null
+
+/datum/gear/proc/spawn_item(var/location, var/metadata, var/mob/living/carbon/human/H)
+	var/datum/gear_data/gd = new(path, location, faction)
+	for(var/datum/gear_tweak/gt in gear_tweaks)
+		if(metadata["[gt]"])
+			gt.tweak_gear_data(metadata["[gt]"], gd, H)
+		else
+			gt.tweak_gear_data(gt.get_default(), gd, H)
+	if(ispath(gd.path, /obj/item/organ/external))
+		var/obj/item/organ/external/external_aug = gd.path
+		var/obj/item/organ/external/replaced_limb = H.get_organ(initial(external_aug.limb_name))
+		replaced_limb.droplimb(TRUE, DROPLIMB_EDGE, FALSE)
+		qdel(replaced_limb)
+	var/item = new gd.path(gd.location)
+	for(var/datum/gear_tweak/gt in gear_tweaks)
+		if(metadata["[gt]"])
+			gt.tweak_item(item, metadata["[gt]"], H)
+		else
+			gt.tweak_item(item, gt.get_default(), H)
+	return item
+
+/datum/gear/proc/spawn_random(var/location)
+	var/datum/gear_data/gd = new(path, location)
+	for(var/datum/gear_tweak/gt in gear_tweaks)
+		gt.tweak_gear_data(gt.get_random(), gd)
+	var/item = new gd.path(gd.location)
+	for(var/datum/gear_tweak/gt in gear_tweaks)
+		gt.tweak_item(item, gt.get_random())
+	return item
+
+/datum/gear/proc/check_species_whitelist(mob/living/carbon/human/H)
+	if(whitelisted && (!(H.species.name in whitelisted)))
+		return FALSE
+	return TRUE
+
+// arg should be a faction name string
```

Comment:
This might be better to fully document, e.g
```suggestion
/** 
  * Compares the character's faction to the tag faction. Returns TRUE if matching. FALSE otherwise.
  *
  * Arguments:
  * * faction_ - String of faction to compare
  */
```